src/utils.py

```diff
@@ -139,12 +141,10 @@ def install_binary_from_url(url: str, sha256_url: str) -> None:
     if sha256_url:
         binary_hash = hashlib.sha256(binary_response.content).hexdigest()
         perform_sha256_checksum(binary_hash, sha256_url)
-    stop_service()
```

Comment:
The reasons for the `stop_service()` here are:
1. The service needs to be stopped before replacing the binary on disk (which is done after this line).
2. To minimize downtime, don't stop the service until the new binary is downloaded to memory (which is done above this line).

So take this in consideration while developing this :+1: